src/allocator.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <cassert>
+#include <mutex>
+#include <memory>
+#include <unordered_set>
+
+#include "fast_logger.h"
+
+struct DefaultAllocator
+{
+	template <class T, class... Args>
+	static T* Allocate(Args&&... args)
+	{
+		return new T(std::forward<Args>(args)...);
+	}
+
+	template <class T>
+	static void Free(T* ptr)
+	{
+		delete ptr;
+	}
+};
+
+struct TrackingAllocator
+{
+	inline static std::mutex Lock;
+	inline static std::unordered_map<void*, const char*> ActiveAllocations;
+	
+	template <class T, class... Args>
+	static T* Allocate(Args&&... args)
+	{
+		auto* ptr = DefaultAllocator::Allocate<T>(std::forward<Args>(args)...);
+
+		{
+			std::unique_lock lock{ Lock };
+			const char* tpn = __FUNCSIG__;
```

Comment:
Maybe std::source_location::current() is better? FUNCSIG is not defined under linux/clang20